File: shared/contracts/typescript/evidenceSource.ts
// EvidenceSource contract
// Represents a single piece of retrieved knowledge that grounded an AI response.
// Attached to summaries, conversation turns, and draft notes.

export enum SourceType {
  ClaimDocument = "CLAIM_DOCUMENT",    // Document attached to the claim
  KnowledgeBase = "KNOWLEDGE_BASE",    // Enterprise knowledge base article
  ClaimNote = "CLAIM_NOTE",            // Prior note on the claim
  PolicyDocument = "POLICY_DOCUMENT",  // Insurance policy language
  Regulation = "REGULATION",           // Regulatory requirement
}

export interface EvidenceSource {
  sourceId: string;
  sourceType: SourceType;
  title: string;
  excerpt: string;         // The specific passage that was used
  relevanceScore: number;  // 0.0 – 1.0, how relevant this chunk was
  documentId: string | null;
  pageReference: string | null; // e.g. "Page 4, Section 2.1"
  retrievedAt: string;     // ISO 8601
}

File: shared/contracts/python/evidence_source.py
"""
EvidenceSource contract — Python / Pydantic.
Represents a single retrieved knowledge chunk that grounded an AI response.
"""

from __future__ import annotations

from datetime import datetime
from enum import Enum

from pydantic import BaseModel, Field


class SourceType(str, Enum):
    CLAIM_DOCUMENT = "CLAIM_DOCUMENT"
    KNOWLEDGE_BASE = "KNOWLEDGE_BASE"
    CLAIM_NOTE = "CLAIM_NOTE"
    POLICY_DOCUMENT = "POLICY_DOCUMENT"
    REGULATION = "REGULATION"


class EvidenceSource(BaseModel):
    source_id: str
    source_type: SourceType
    title: str
    excerpt: str
    relevance_score: float = Field(ge=0.0, le=1.0)
    document_id: str | None = None
    page_reference: str | None = None
    retrieved_at: datetime
